frontend/components/utils/FormatCurrency.vue

```diff
@@ -1,10 +1,13 @@
 <template>
-    <span>
-        {{ sign }}<animated-number v-if="isValidNumber" :value="value" :decimal-places="decimals" /><span
-            v-else-if="value"
-            >NaN</span
-        ><span class="uppercase"> {{ currency }}</span>
-    </span>
+    <span
+        >{{ sign
+        }}<span v-if="isValidNumber"
+            ><animated-number v-if="value > 0.00001 || value === 0" :value="value" :decimal-places="decimals" /><span
```

Comment:
Can you please move `value > 0.00001 || value === 0` into a computed property and reuse it in the `decimals` (to not compute log and to avoid checking against infinity)? Otherwise, you can reuse `decimals` here as `decimals > 5` instead of creating a separate logic for the same thing

Reply:
We may also want to check `if (number > 0)` before counting decimals and applying log10

Reply:
One last thing: we may want to do it inside the animated number component (taking into account decimals prop as minimum), so this logic is applied everywhere

Reply:
I understand moving `value > 0.00001 || value === 0` into a computed property, but I do not understand why I should use it in the `decimals` calculation to replace the check against Infinity.

Reply:
Then, let's do it the other way around and use `decimals` inside the `isValidNumber`?

Reply:
I got to be honest. I still don't see the point in this. I would be up for a quick call if that makes the explanation easier.